pkg/substitution/substitution.go

```diff
@@ -234,7 +234,8 @@ func extractExpressionFromString(s, prefix string) (string, bool) {
 	return match[0], true
 }
 
-func extractVariablesFromString(s, prefix string) ([]string, bool, string) {
+// ExtractVariablesFromString extracts variables from an input string s with the given prefix via regex matching.
+func ExtractVariablesFromString(s, prefix string) ([]string, bool, string) {
```

Comment:
this function is now exported so we need to add unit tests for it

Reply:
Good catch. I will do that.

Reply:
Done now!!